app/src/main/java/com/hover/stax/domain/use_case/auth/AuthUseCase.kt

```diff
@@ -0,0 +1,30 @@
+package com.hover.stax.domain.use_case.auth
+
+import com.hover.stax.domain.model.Resource
+import com.hover.stax.domain.model.StaxUser
+import com.hover.stax.domain.repository.AuthRepository
+import com.hover.stax.domain.repository.StaxUserRepository
+import com.hover.stax.utils.TokenUtils
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import timber.log.Timber
+
+class AuthUseCase(private val authRepository: AuthRepository, private val staxUserRepository: StaxUserRepository) {
+
+    fun authorize(idToken: String): Flow<Resource<StaxUser>> = flow {
```

Comment:
My initial suggestion would eliminate doing this implementation on the UseCase

Reply:
Yeah, I just noticed how strange this is... You would put it in the repository?